process/block/metablock.go

```diff
@@ -1487,6 +1487,7 @@ func (mp *metaProcessor) updateState(lastMetaBlock data.MetaHeaderHandler, _ []b
 	)
 
 	mp.setFinalizedHeaderHashInIndexer(lastMetaBlock.GetPrevHash())
+	mp.blockChain.SetHighestFinalBlockCoordinates(prevMetaBlock.GetNonce(), prevMetaBlockHash, prevMetaBlock.GetRootHash())
```

Comment:
this is not exactly the final block it is the prev to final block.

Reply:
should use `lastMetaBlock`

Reply:
so should set then lastMetaBlock.GetNonce(), lastMetaBlockHash can be used from param list (un-ignore) and lastMetaBlock.GetRootHash()

Reply:
Fixed.